components/eam/src/physics/crm/cospsimulator_intr.F90

```diff
@@ -0,0 +1,3184 @@
+       nht_cosp           ! Number of height for COSP radar and calipso simulator outputs.  
+                          !  *set to 40 if csat_vgrid=.true., else set to Nlr*
+  
+  ! ######################################################################################
+  ! Bin-boundaries for mixed dimensions. Calculated in cospsetupvales OR in cosp_config.F90
+  ! ######################################################################################
+  real(r8), target :: prsmid_cosp(nprs_cosp)            ! pressure midpoints of COSP ISCCP output
+  real(r8), target :: prslim_cosp(2,nprs_cosp)
+  real(r8), target :: taumid_cosp(ntau_cosp)            ! optical depth midpoints of COSP ISCCP output
+  real(r8), target :: taulim_cosp(2,ntau_cosp)
+  real(r8), target :: srmid_cosp(nsr_cosp)              ! sr midpoints of COSP lidar output     
+  real(r8), target :: srlim_cosp(2,nsr_cosp)
+  real(r8), target :: sza_cosp(nsza_cosp)
+  real(r8), target :: dbzemid_cosp(CLOUDSAT_DBZE_BINS)          ! dbze midpoints of COSP radar output
+  real(r8), target :: dbzelim_cosp(2,CLOUDSAT_DBZE_BINS)
+  real(r8), target :: htmisrmid_cosp(nhtmisr_cosp)      ! htmisr midpoints of COSP misr simulator output
+  real(r8), target :: htmisrlim_cosp(2,nhtmisr_cosp)
+  real(r8), target :: taumid_cosp_modis(ntau_cosp_modis)! optical depth midpoints of COSP MODIS output
+  real(r8), target :: taulim_cosp_modis(2,ntau_cosp_modis)
+  real(r8), target :: reffICE_binEdges_cosp(2,numMODISReffIceBins)
+  real(r8), target :: reffLIQ_binEdges_cosp(2,numMODISReffLiqBins)
+  real(r8), target :: reffICE_binCenters_cosp(numMODISReffIceBins)
+  real(r8), target :: reffLIQ_binCenters_cosp(numMODISReffLiqBins)
+
+  real(r8) :: htmlmid_cosp(nhtml_cosp)                     ! Model level height midpoints for output
+  real(r8),allocatable, target :: htlim_cosp(:,:)          ! height limits for COSP outputs (nht_cosp+1)
+  real(r8),allocatable, target :: htmid_cosp(:)            ! height midpoints of COSP radar/lidar output (nht_cosp)
+  integer, allocatable, target :: scol_cosp(:)             ! sub-column number (nscol_cosp)
+
+  ! ######################################################################################
+  ! Default namelists
+  ! The CAM and COSP namelists defaults are set below.  Some of the COSP namelist 
+  ! variables are part of the CAM namelist - they all begin with "cosp_" to keep their 
+  ! names specific to COSP. I set their CAM namelist defaults here, not in namelist_defaults_cam.xml
+  !  Variables identified as namelist variables are defined in
+  !  ../models/atm/cam/bld/namelist_files/namelist_definition.xml
+  ! ######################################################################################
+  ! CAM
+  logical :: cosp_amwg             = .false. ! CAM namelist variable default, not in COSP namelist
+  logical :: cosp_lite             = .false. ! CAM namelist variable default, not in COSP namelist
```

Comment:
So "CAM" variables that have "cosp" in the name?
I think we can remove or edit the "CAM" section header here, and delete the redundant line comments with  "CAM namelist variable default"

Reply:
The whole namelist handling here really needs to be revamped I think. There's a lot of redundant and conflicting options in here that overwrite others.

Reply:
Note that revamping this would require also revamping the non-MMF version, because they share namelist options. We can certainly do this, but might be better to do in a subsequent PR for expediency?

Reply:
Agreed.